### ISSUE TITLE
Warn when sharing a private post externally

### ISSUE THREAD

**Issue by Klikini** [mastodon/mastodon#31200]
### Pitch

When clicking "share", "embed", or "copy link to post" on a post that is set to "followers only" or "mentioned users only", a warning should be displayed stating that the recepient of the link might not be able to see the post.

I could probably implement this and open a PR myself if I get confirmation that it would be a welcome change.

### Motivation

I sent a link to a post that I didn't notice was private to some friends and it took a while to figure out why they were seeing a 404.

In hindsight, I now know to check first (if I remember to!), but showing a warning in this case (like Bluesky does) would have avoided this situation entirely.

**Comment by ClearlyClaire** [comment]
Hi! We should not be offering “Share” and “Embed” for “followers only” and “mentioned users only” posts. Where have you seen that?

We do offer “Copy link to post”, and I think we should continue doing so. I'm afraid adding a warning would be pretty annoying, but it's definitely worth considering.

**Comment by Klikini** [comment]
You're right, I went back to look at the UI while writing my post and I didn't realize I was looking at a public post with share/embed options. That's fine as-is :)

I still think the warning on "copy link" would be good to have, and it could be an account preference individual users could choose to disable if they find it annoying.
